src/darkleaf/di/core.clj

```diff
@@ -131,12 +131,16 @@
               [nil :required] (missing-dependency! stack)
               (recur tail (assoc built-map key obj)))))))))
 
-(defn- try-run [proc]
+(defn- handle-ex [proc handler]
```

Comment:
@KGOH @krevedkokun  it seems ugly for me. Do you know any other solution? 
In Java we should not catch Errors, but in Clojure I want to catch and recover on AssertionError.

Reply:
```
(defn- try-run [proc]
  (try (proc)
       nil
       (catch Exception e e)
       (catch AssertionError e e)))

(defn- try-build [ctx key]
  (try (build ctx key)
       (catch Exception e e (??? e))
       (catch AssertionError e e (??? e))))

```